pkg/api/meta/meta.go

```diff
@@ -373,7 +414,8 @@ type genericAccessor struct {
 	labels            *map[string]string
 	annotations       *map[string]string
 	// TODO: This has to be a pointer to list, otherwise we can implement SetOwnerReferences
-	ownerReferences []unversioned.ObjectReference
+	// ownerReferences []*unversioned.ObjectReference
+	ownerReferences listAccessor
```

Comment:
listAccessor seems to convert everything to []interface{} which is going to mean a lot of unnecessary copying. I'm fine if you make listAccessor itself use a reflect.Value and work with a list of anything.